pymc_marketing/mlflow.py

```diff
@@ -577,8 +941,35 @@ def new_fit(*args, **kwargs):
                 "saturation_name",
                 json.loads(idata.attrs["saturation"])["lookup_name"],
             )
+
+            # Align with the default values in pymc.sample
+            tune = kwargs.get("tune", 1000)
+
+            if log_sampler_info:
+                log_sample_diagnostics(idata, tune=tune)
+                log_arviz_summary(
+                    idata,
+                    "summary.html",
+                    var_names=summary_var_names,
+                    **arviz_summary_kwargs,
+                )
+
             log_inference_data(idata, save_file="idata.nc")
 
+            if log_loocv:
+                log_loocv_metrics(idata=idata)
+
+            posterior_preds = self.sample_posterior_predictive(self.X)
+            log_summary_metrics(
+                y_true=self.y,
+                y_pred=posterior_preds[self.output_var],
+            )
+
+            log_model(
+                self,
+                registered_model_name=registered_model_name,
+            )
```

Comment:
I conceptually like the idea of this being included in the autologging but the PyMC models are a bit tough. I lean a bit more to keeping this separate because someone's model flow might be: 

```python
with mlflow.start_run() as run: 
    mmm.fit(X, y)
    mmm.sample_posterior_predictive(X)
    mmm.sample_posterior_predictive(X_new, predictions=True)
    load_model(mmm, registered_model_name="my-registered-model")
```

So having this on `fit` is a bit restrictive. But I think the signature is easy enough to throw in where you want to!

Reply:
Ah yeah, I see what you mean - I've removed it from the fit patch ☺️

Reply:
All good, just removed it ☺️